perfkitbenchmarker/providers/profitbricks/profitbricks_virtual_machine.py

```diff
@@ -0,0 +1,256 @@
+            self.dc_id, self.dc_status = util.CreateDatacenter(self.header,
+                                                               self.location)
+
+            # Create LAN so that we can utilize a public IP address
+            self.lan_id, self.lan_status = util.CreateLan(self.header,
+                                                          self.dc_id)
+
+            # Switch to false as we now have our DC and LAN to work with
+            self.need_dc = False
+
+        # Grab ssh pub key to inject into new VM
+        with open(self.ssh_public_key) as f:
+            public_key = f.read().rstrip('\n')
+
+        # Find an Ubuntu image that matches our location
+        self.image = util.ReturnImage(self.header, self.location)
+
+        # Find necessary specs for the user's machine_type selection
+        flavor_ram, flavor_cores = util.ReturnFlavor(self.machine_type)
+
+        if not self.ram:
+            self.ram = flavor_ram
+        if not self.cores:
+            self.cores = flavor_cores
+
+        # Create server POST body
+        new_server = {
+            'properties': {
+                'name': self.name,
+                'ram': self.ram,
+                'cores': self.cores,
+                'availabilityZone': self.zone
+            },
+            'entities': {
+                'volumes': {
+                    'items': [
+                        {
+                            'properties': {
+                                'size': self.disk_size,
+                                'name': 'volume1',
+                                'image': self.image,
+                                'bus': 'VIRTIO',
+                                'type': self.profitbricks_disk_type,
+                                'sshKeys': [public_key]
+                            }
+                        }
+                    ]
+                },
+                'nics': {
+                    'items': [
+                        {
+                            'properties': {
+                                'name': 'nic1',
+                                'lan': self.lan_id
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+
+        # Build Server URL
+        url = '%s/datacenters/%s/servers' % (PROFITBRICKS_API, self.dc_id)
+
+        # Provision Server
+        r = util.PerformRequest('post', url, self.header, json=new_server)
+        logging.info('Creating server: %s' % self.name)
+
+        # Parse Required values from response
+        self.server_status = r.headers['Location']
+        response = r.json()
+        self.server_id = response['id']
+
+        # The freshly created server will be in a locked and unusable
+        # state for a while, and it cannot be deleted or modified in
+        # this state. Wait for the action to finish and check the
+        # reported result.
+        if not util.WaitFor(self.server_status, self.header):
```

Comment:
In the other providers, vms are not immediately available, either, but pkb [uses the `Exists`](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/resource.py#L101) method to do it's own waiting for them to be ready.  For VMs, even after they are created, there is [a wait for them to be booted](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/benchmark_spec.py#L353).  So I don't think this check is necessary.